File: examples/leetcode-out/zig/12/integer-to-roman.zig
const std = @import("std");

fn intToRoman(num: i32) []const u8 {
	const values = &[_]i32{@as(i32,@intCast(1000)), @as(i32,@intCast(900)), @as(i32,@intCast(500)), @as(i32,@intCast(400)), @as(i32,@intCast(100)), @as(i32,@intCast(90)), @as(i32,@intCast(50)), @as(i32,@intCast(40)), @as(i32,@intCast(10)), @as(i32,@intCast(9)), @as(i32,@intCast(5)), @as(i32,@intCast(4)), @as(i32,@intCast(1))};
	const symbols = &[_]i32{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
	var result = "";
	var i = @as(i32,@intCast(0));
	while ((num > @as(i32,@intCast(0)))) {
		while ((num >= values[i])) {
			result = (result + symbols[i]);
			num = (num - values[i]);
		}
		i = (i + @as(i32,@intCast(1)));
	}
	return result;
}

pub fn main() void {
}

File: examples/leetcode-out/java/12/integer-to-roman.java
public class Main {
  static String intToRoman(int num) {
    Object[] values = new int[] {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
    Object[] symbols =
        new String[] {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
    String result = "";
    int i = 0;
    while ((num > 0)) {
      while ((num >= values[i])) {
        result = (result + symbols[i]);
        num = (num - values[i]);
      }
      i = (i + 1);
    }
    return result;
  }

  static void test_example_1() {
    expect((intToRoman(3) == "III"));
  }

  static void test_example_2() {
    expect((intToRoman(58) == "LVIII"));
  }

  static void test_example_3() {
    expect((intToRoman(1994) == "MCMXCIV"));
  }

  static void test_small_numbers() {
    expect((intToRoman(4) == "IV"));
    expect((intToRoman(9) == "IX"));
  }

  public static void main(String[] args) {
    test_example_1();
    test_example_2();
    test_example_3();
    test_small_numbers();
  }

  static void expect(boolean cond) {
    if (!cond) throw new RuntimeException("expect failed");
  }
}
